activestorage/app/models/active_storage/blob.rb

```diff
@@ -331,8 +331,9 @@ def service
   private
     def compute_checksum_in_chunks(io)
       raise ArgumentError, "io must be rewindable" unless io.respond_to?(:rewind)
+      return unless service.try(:checksum_algorithm)
 
-      ActiveStorage.checksum_implementation.new.tap do |checksum|
+      ActiveStorage::Checksum.for(service.try(:checksum_algorithm)).new.tap do |checksum|
```

Comment:
I'd rather avoid `try` here, it's a big smell.

Reply:
Removed in favor of checking if `service` is nil